index.js

```diff
@@ -1,23 +1,193 @@
+function Board(players) { 
+  this.players = players; // just a placeholder for now - on merge should combine 
+  this.spaces = [];
+} 
+
+// class definitions for any space on the board - 
+// properties, utilities, free parking, go, etc. 
+function Space(name) { 
+  this.name = name; 
+  this.occupier = null;
+}
+// every space needs a landing action
+Space.prototype.perform_landing_action = function(){
+  console.log("You landed on " + this.name);
+}; 
+
+function Property(name, cost, property_group) { 
+  Space.call(this, name);
+  this.cost = cost;
+  this.mortage = .5*cost;
+  this.property_group = property_group; // "red", "blue", "railroad", etc.
+  this.owner = null;
+}
+Property.prototype = Object.create(Space.prototype); // subclassing space
+
+// rent should be array with following format: 
+// [1 owned, 2 owned ... (3 owned, 4 owned)]
+function UtilityProperty(name, cost, property_group, rent) { 
+  this.rent = rent;
+  Property.call(this, this.name, this.cost, this.property_group);
+}
+UtilityProperty.prototype = Object.create(Property.prototype);
+UtilityProperty.prototype.perform_landing_action = function(player) { 
+}; 
+
+// rent should be array with following format: 
+// [rent, 1 house, 2 houses, 3 houses, 4 houses, hotel] 
+function HousingProperty(name, cost, property_group, rent) { 
+  this.rent = rent;
+  Property.call(this, this.name, this.cost, this.property_group);
+}
+HousingProperty.prototype = Object.create(Property.prototype);
+HousingProperty.prototype.perform_landing_action = function(player) { 
 };
 
-module.exports = test;
\ No newline at end of file
+
+function Chance() {} 
+Chance.prototype = Object.create(Space.prototype);
+Chance.prototype.perform_landing_action = function(player) { 
+}; 
+
+function CommunityChest() {}
+CommunityChest.prototype = Object.create(Space.prototype);
+Chance.prototype.perform_landing_action = function(player) { 
+}; 
+
+function FreeParking() {}
+FreeParking.prototype = Object.create(Space.prototype);
+FreeParking.prototype.perform_landing_action = function(player) { 
+}; 
+
+function Jail() {}
+GoToJail.prototype = Object.create(Space.prototype);
+GoToJail.prototype.perform_landing_action = function(player) { 
+}; 
+
+function GoToJail() {}
+GoToJail.prototype = Object.create(Space.prototype);
+GoToJail.prototype.perform_landing_action = function(player) { 
+}; 
+
+function Tax() { 
+  this.tax_amount = 0; // give default value of 0, should be overridden 
+}
+Tax.prototype = Object.create(Space.prototype);
+Tax.prototype.perform_landing_action = function(player) { 
+  this.player.money -= this.tax_amount;
+}; 
+
+function IncomeTax() { 
+  this.tax_amount = 200;
+}
+IncomeTax.prototype = Object.create(Tax.prototype);
+
+function SuperTax() { 
+  this.tax_amount = 100;
+}
+SuperTax.prototype = Object.create(Tax.prototype);
+
+function Go() {}
+Go.prototype = Object.create(Space.prototype);
+Go.prototype.perform_landing_action = function(player) { 
+  this.player.money += 400;
```

Comment:
The turn logic is currently handling testing if you go past go. Double money on Go is a house rule, but if we want to include it this should be += 200 (since we'll have the other 200 for passing go).